### ISSUE TITLE
shift->next::method  broken for 5.22

### ISSUE THREAD

**Issue by toddr** [rurban/perl-compiler#298]
Based on https://github.com/toddr/perl/blob/blead/t/mro/complex_c3.t

The program:

``` perl
#!./perl
package D;
sub testmeth { "wrong" }

package C;
our @ISA = qw/D/;
sub testmeth { "right" }

package B;
our @ISA = qw/D/;

package A; use mro 'c3';
our @ISA = qw/B C/;
sub testmeth { shift->next::method }

A->testmeth() eq "right" and print "ok\n"
```

I've simplified the inheritance tree to this:
```
                            ---          |
Level 2                  5 | D |         |
                            ---          |
                          /  |           |
                         /   |           |
                        /    |           |
                       /     |           |
                      /      |           |
                  ---       ---          |
Level 1        1 | B |   2 | C |         |
                  ---       ---          |
                    \       /            |
                     \     /             |
                       ---               v
Level 0             0 | A |            (More Specialized)
                       ---
```

$>perlcc -e 'package D; sub testmeth { "wrong" } package C; our @ISA = qw/D/; sub testmeth { "right" } package B; our @ISA = qw/D/; package A; use mro "c3"; our @ISA = qw/B C/; sub testmeth { shift->next::method } A->testmeth() eq "right" and print "ok\n"'; ./a.out

zsh: segmentation fault  ./a.out

**Comment by toddr** [comment]
This passes on 5.20

**Comment by toddr** [comment]
broken on 5.21.11 for what that matters.

**Comment by atoomic** [comment]
testc.sh 298

**Comment by toddr** [comment]
```diff
commit 1a33a0598e4c684205d292afcb97de6d79d17e7d
Author: Father Chrysostomos <sprout@cpan.org>
Date:   Sun Aug 18 00:36:05 2013 -0700

    Move super cache into mro meta
    
    Iterated hashes shouldn’t have to allocate space for something
    specific to stashes, so move the SUPER method cache from the
    HvAUX struct (which all iterated hashes have) into the mro
    meta struct (which only stashes have).
```

**Comment by rurban** [comment]
See branch `mro-298`

**Comment by toddr** [comment]
While these no longer segfault after 1dbeb490d85ccc9903bf86a89b70d67db068e338, testc.sh 298 continues to not pass.

**Comment by toddr** [comment]
Example failure right now:

# Can't locate object method "method" via package "next" at mro/complex_c3_utf8.t line 86.

**Comment by rurban** [comment]
Fixed in branch `no-mro-gh314`  00be98f

**Comment by rurban** [comment]
Should be fixed now with 0819c33033bdd7a70e03499b7c5e87306774b505
